tools/tpm2_createak.c

```diff
@@ -556,6 +563,17 @@ int tpm2_tool_onrun(TSS2_SYS_CONTEXT *sapi_context, tpm2_option_flags flags) {
         return 1;
     }
 
+    if (ctx.find_persistent_ak) {
+        bool ret = tpm2_capability_find_vacant_persistent_handle(sapi_context,
+                        &ctx.ak.in.handle);
+        if (!ret) {
+            LOG_ERR("ak-handle/k passed with a value of '-' but unable to find"
+                    " a vacant persistent handle!");
+            return 1;
+        }
+        tpm2_tool_output("AK persistent-handle: 0x%x\n", ctx.ak.in.handle);
```

Comment:
maybe `ak-persistent-handle: 0x1234`

I'm not a fan of spaces, even though JSON handles it fine. I might not have told you, but all outputs are in JSON, this way tools with compound outputs can have some sane way of parsing them.

Reply:
I hadn't realised that we were outputting JSON with a consideration to parsing the tools output. With that in mind particularly, I like your proposal.

Reply:
Even for tools that have a single output, it's considering a "JSON Scalar" at that point. So the JSON output is quite flexible.

Reply:
To be clear, in your case it's a JSON key value pair.